zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java

```diff
@@ -225,6 +226,9 @@ private void assignAuthInfo(ServerCnxn cnxn, String clientId, Set<String> domain
       // "super" scheme gives access to all znodes without checking znode ACL vs authorized domain name
       commonSuperUserDomains.stream().forEach(d ->
           newAuthIds.add(new Id(X509AuthenticationUtil.SUPERUSER_AUTH_SCHEME, d)));
+      if (X509AuthenticationConfig.getInstance().isStoreAuthedClientIdEnabled()) {
```

Comment:
this means that even helix or tools or any other applications under cross domain list will have client ids in authinto. Just wanted to clarify, you are adding this to support d2 clients who are talking mapped to helix domain?

Reply:
Good point! Synced up with @rahulrane50 offline, clients belong to superUserDomains won't need clientId since it already has super user access, so we don't need to worry about it losing access to the nodes that is under clientId of another application which belongs to the same domain. Will remove this line.